### ISSUE TITLE
Invert if

### ISSUE THREAD

**Issue by ElemarJR** [code-cracker/code-cracker#88]
This is essentially a refactoring, so we are classifying it as a `Hidden` diagnostic.

**Comment by giggio** [comment]
@elemarjr can you post an example of exactly what you mean? Would be in case of an `if`...`else`?

**Comment by Kavignon** [comment]
essentially, what it means to invert a if would be something like this @giggio 
//not inverted
public string Method(Object obj)
{
    if(obj != null)
    { 
         return obj.ToString();
     }
}

public string SecondMethod(Object obj)
{
    if(obj == null) return null;
    return obj.ToString();
}

**Comment by Kavignon** [comment]
that would, indeed, be a good code refactoring @ElemarJR 
The big impact of this refactoring is that it avoids the need of the nested if structure @giggio

**Comment by giggio** [comment]
@Kavignon I have updated the description, see if you agree.
@ElemarJR @carloscds I need some of you to validate it.

I liked this one, I don't have an issue assigned to me at this moment, I would like to work on this one.

**Comment by Kavignon** [comment]
I agree with your update ! 
I'm still working on #88.
But i'd like to take that one on as soon as I'm done with it !

**Comment by giggio** [comment]
@code-cracker/owners I need you to validate this issue. It seems `Ready` to me.

**Comment by carloscds** [comment]
@giggio For me it´s ok.

**Comment by giggio** [comment]
Ok, it is `Ready`. @Kavignon finish you item first, then you can get this one. :)

**Comment by gandarez** [comment]
I'm working on it. Finishing up with writing tests.

**Comment by gandarez** [comment]
Clarifying: You mean a simple statement-inversion function or a way to reduce nesting? What I've done so far is a simple statement-inversion function.

**Comment by giggio** [comment]
@Kavignon Just an inversion. See the tests on the description, they should pass. Reducing nesting comes for free. :)

**Comment by gandarez** [comment]
There're two scenarios. One of them is simply invert a full if statement (if ... else). The other one is to reduce nesting which in my opinion differs from the first one. Do you agree?
The simple scenario I already covered.

**Comment by giggio** [comment]
I mentioned @Kavignon incorrectly. Sorry about that. @gandarez is working on it.

**Comment by giggio** [comment]
@gandarez I don't agree. The diference is that the second one returns. The first one doesnt.

**Comment by gandarez** [comment]
@giggio Ok i'll make effort to make it as described.

**Comment by gandarez** [comment]
I'm still working on it. A little bit belated because I'm stuck at a point.

**Comment by giggio** [comment]
Ok, if you need help let us know, we can help you.

**Comment by giggio** [comment]
@gandarez How is this coming?
